### PR TITLE
Porting attach command(WIP)

### PR DESCRIPTION
**Pull request by Rocketct**

### CODE REVIEW

File: daemon/client/client.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func main() {
 	}
 
 	install()
-	install()
+	//install()
```
Comment (cmaglie) [inline daemon/client/client.go]: why this change?

File: daemon/client/client.go [hunk 2]
```diff
@@ -194,6 +194,33 @@ func main() {
 	fmt.Printf("---> %+v\n", details)
 	fmt.Println()
 
+	// BOARDS ATTACH
+	fmt.Println("=== calling BoardAttach(serial:///dev/ttyACM0)")
+	boardattachresp, err := client.BoardAttach(context.Background(), &rpc.BoardAttachReq{
+		Instance:   instance,
+		BoardURI:   "/dev/ttyACM0",
+		SketchPath: "/home/riccardo/Arduino/MyFirstSketch",
```
Comment (cmaglie) [inline daemon/client/client.go]: this path is wrong, use the command line argument

File: rpc/board.proto [hunk 3]
```diff
@@ -51,3 +51,15 @@ message RequiredTool {
...
+message BoardAttachReq {
+  Instance instance = 1;
+  string boardURI = 2;
+  string sketchPath = 3;
+  string boardFlavour = 4;
+  string searchTimeout = 5;
+}
+
+message BoardAttachResp {
+  TaskProgress TaskProgress = 1;
```
Comment (cmaglie) [inline rpc/board.proto]: in proto files we do not use camel case but snake case:
`boardURI` -> `board_uri`
`sketchPath` -> `sketch_path`
`boardFlavour` -> `board_flavour`
`searchTimeout` -> `search_timeout`
`TaskProgress` -> `task_progress`

File: commands/board/attach.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+		board := findBoardFunc(pm, monitor, deviceURI)
+		if board == nil {
+			return nil, fmt.Errorf("no supported board found at %s", deviceURI.String())
+		}
+		taskCB(&rpc.TaskProgress{Name: "Board found: " + board.Name(), Completed: true})
+
+		sketch.Metadata.CPU = sketches.BoardMetadata{
+			Fqbn: board.FQBN(),
+			Name: board.Name(),
+		}
```
Comment (cmaglie) [inline commands/board/attach.go:100]: We should probably stop the monitor: when running as a pure CLI it doesn't matter because when the CLI exits the monitor is released by the OS, but when running as a daemon I don't know if the resource is leaked.

Please put a `TODO` this is related to board discovery and `board list` command.

**Review by cmaglie — APPROVED** [review]
Some small changes here and there but overall is good.

File: daemon/client/client.go (hunk 1 above)
Comment (Rocketct) [reply daemon/client/client.go]: only test i forgot to leave  the comment.

File: daemon/client/client.go (hunk 2 above)
Comment (Rocketct) [reply daemon/client/client.go]: ok

File: rpc/board.proto (hunk 3 above)
Comment (Rocketct) [reply rpc/board.proto]: ok, done

File: commands/board/attach.go (hunk 4 above)
Comment (Rocketct) [reply commands/board/attach.go:100]: done
